lessons/git/ci-github/lesson.md

```diff
@@ -0,0 +1,128 @@
+be generated before pushing to the live website.
+
+## Why should you use it?
+
+There are [many reasons](https://about.gitlab.com/2015/02/03/7-reasons-why-you-should-be-using-ci/)
+to use CI, especially if your codebase is large and complex, when multiple people 
+work on the code, when it is relied on by others, or if it is for production
+(industry or commercial) purposes. If your code is small or simple or a one-time 
+thing, it is often not necessary to use CI. BUT! Using CI will make sure that your 
+code is robust and more likely to be accurate and correct... which is especially
+important in most if not all scientific fields.
+
+## Travis and the `.travis.yml` file
+
+As I mentioned, you use travis for GitHub. In order to use travis, you need to
+sign up and follow the [getting started](https://docs.travis-ci.com/user/getting-started/)
+steps. Travis has many supported [languages](https://docs.travis-ci.com/),
+including [R](https://docs.travis-ci.com/user/languages/r/) and 
+[Python](https://docs.travis-ci.com/user/languages/python/). 
+
+The two main things you need for travis are to include a `.travis.yml` file in
+your GitHub repo and link travis to it via the [travis website](https://travis-ci.org/).
+
+### Example R `.travis.yml` file
+
+```
+language: r
+cache: packages # to reduce the build time for dependent packages
+r:
+    - release
+    - devel
+script:
+    # If building packages
+    - R CMD build .
+    - R CMD check *.tar.gz
+    # If running from an R or shell file
+    - Rscript test.R
+    # or:
+    - sh ./deploy.sh
+```
+
+Package dependencies are by default searched for in the 
+[`DESCRIPTION`](http://r-pkgs.had.co.nz/description.html) file. Otherwise,
+specific packages can be installed by including this in the `.travis.yml` file:
+
+```
+r_packages:
+    - "packagename"
+```
+
+If you are creating packages and want to use travis to test your packages, use
+the R command `devtools::use_travis()` to get you started.
+
+Building websites via R Markdown is relatively straight forward. You first need
+to create a `deploy.sh` file of some sort (see 
+[this file](https://github.com/codeasmanuscript/rworkshops/blob/master/deploy.sh) as
+an example) and include travis commands to push to a `gh-pages` branch (see
+[this file](https://github.com/codeasmanuscript/rworkshops/blob/master/.travis.yml) as
+an example). For a more detailed example and walkthrough, see 
+[this Gist](https://gist.github.com/willprice/e07efd73fb7f13f917ea.html).
+
+### Example Python `.travis.yml` file
+
+```
+language: python
+python:
+    - "3.6"
+# install dependencies
+install:
+    - pip install -r requirements.txt # if you put your dependencies in a file
+script: pytest
+```
+
+I am not nearly as familiar with Python, however, the same things that apply to R
+also apply to Python. You need to install dependencies and use the `script` option
+to specific what commands or file to run.
```

Comment:
yeah sorry that's not what I meant, I was thinking it would be good to just note that people don't need to include the common python testing packages which come with Travis. so yes, I agree, for sure we should have the line showing the requirements.txt